api/spec/v1.yaml

```diff
@@ -1500,6 +1500,15 @@ components:
         message:
           type: string
           description: The message of a failed transaction.
+        params:
```

Comment:
We probably need another field for the _name_ (type) of the error? Analogous to `evm_fn_name` (which we're missing) and `evm_fn_params`.
It's unfortunate that we're creating so much boileplate and fields for something so small ... but I don't think it's worth taking shortcuts here. By contrast, in the DB, we _could_ (ab)use the first field of `error_params` to store just the plain-string name of the error type. If you go with that (instead of adding yet another column to the table), please be sure to update the docstring.

Reply:
Hm we currently we return the error name in the `TxError.message` field, which is otherwise empty. So a custom error like `InsufficientBalance(threshold uint64)` would be returned as
```
TxError.message          | TxError.params
__________________________________
InsufficentBalance       | [{name: "threshold", type: uint64, value: 100}]
```

However, one quirk is that default revert errors with a format`Error(string)`, e.g., `revert("insufficient balance")` are returned as
```
TxError.message          | TxError.params
__________________________________
insufficient balance     |  
```

whereas if we were strictly consistent with formatting as the other error types, it 'should' look like
```
TxError.message          | TxError.params
__________________________________
Error                             | [{name: "", type: string, value: "insufficient balance"}]
```

Imo the former case is easier to understand, so I'd be okay with keeping it as-is. Also, the `revert("")`/`Error(string)` error types are implicit, which makes me a bit more willing to special case this type. I'd imagine that actual user-defined error types in the abi would have more descriptive error names/types, which would be more appropriate to return in the error_message field than just `Error`.

Reply:
Addendum to the above^:

We should include the `reverted: ` prefix for all abi error names as well. Updated the descriptions accordingly

Reply:
Thank you for fixing the `reverted: ` prefix!

I still think that in an ideal world, the error type (e.g. `InsufficientBalance`) should be stored and reported in a separate field like `revert_error_type`. `revert_error_type` and `revert_params` are the machine-readable version of the revert reason, whereas `message` is the human-readable version.

That said, let's just wrap this up for now and be pragmatic. It's reasonably easy and reliable to extract the type from `message`, and I don't see users needing this precise breakdown very often.

**The one thing I still would like to change** now that I understand the behavior is to make `message` indeed the human-readable version of the full error. Which IMO would be something like `reverted: <error_type>(<arg1_value>, <arg2_value>, ...)`. For the example above, it would be `reverted: InsufficientBalance(100)`. This will be very handy for the Explorer too, because presumably this is the value they'll want to show by default.

LMK if you disagree, otherwise #shipit! :+1:

Reply:
Good call, added!